sql_extensions/migrations/special-queries.sql

```diff
@@ -47,15 +47,54 @@ WHERE
 ORDER BY
     (data -> 'market_metadata' ->> 'market_id')::NUMERIC,
     (data ->> 'transaction_version')::NUMERIC DESC,
-    (data ->> 'event_index')::NUMERIC DESC;
+    (data -> 'state_metadata' ->> 'market_nonce')::NUMERIC DESC;
 
 CREATE OR REPLACE FUNCTION UPDATE_LATEST_STATE()
 RETURNS TRIGGER AS $$
 BEGIN
-  DELETE FROM inbox_latest_state
-  WHERE (inbox_latest_state.data->'market_metadata'->>'market_id')::numeric = (NEW.data->'market_metadata'->>'market_id')::numeric
-  AND (inbox_latest_state.data->'state_metadata'->>'bump_time')::numeric <= (NEW.data->'state_metadata'->>'bump_time')::numeric;
-  INSERT INTO inbox_latest_state SELECT NEW.*;
+  IF EXISTS (
+    SELECT 1
+    FROM inbox_latest_state
+    WHERE (inbox_latest_state.data->'market_metadata'->>'market_id')::numeric = (NEW.data->'market_metadata'->>'market_id')::numeric
+      AND (inbox_latest_state.data->'state_metadata'->>'market_nonce')::numeric < (NEW.data->'state_metadata'->>'market_nonce')::numeric
+  ) THEN
+    RETURN NEW;
+  END IF;
+
+  INSERT INTO inbox_latest_state (
+    sequence_number,
+    creation_number,
+    account_address,
+    transaction_version,
+    transaction_block_height,
+    type,
+    data,
+    inserted_at,
+    event_index,
+    indexed_type,
+    market_id
+  )
+  VALUES (
+    NEW.sequence_number,
+    NEW.creation_number,
+    NEW.account_address,
+    NEW.transaction_version,
+    NEW.transaction_block_height,
+    NEW.type,
+    NEW.data,
+    NEW.inserted_at,
+    NEW.event_index,
+    NEW.indexed_type,
+    (NEW.data -> 'market_metadata' ->> 'market_id')::numeric
+  )
+  ON CONFLICT (market_id) DO UPDATE SET
+    transaction_version = EXCLUDED.transaction_version,
+    transaction_block_height = EXCLUDED.transaction_block_height,
+    data = EXCLUDED.data,
+    inserted_at = EXCLUDED.inserted_at,
+    event_index = EXCLUDED.event_index
+  WHERE (inbox_latest_state.data->'state_metadata'->>'market_nonce')::numeric < (EXCLUDED.data->'state_metadata'->>'market_nonce')::numeric;
```

Comment:
Perhaps there's a better way to do this, but I don't know if there's a "spread" operator or something. As far as I know, there's not.

But basically this fails constantly if there's no `market_id`, so we need to add it when we insert the new data, because the original `::State` event that triggers it doesn't have it as a top-level column, it's only a nested jsonb field.

The on conflict with the `WHERE` clause is to ensure we aren't inserting stale data, although I'm not fully